ruby/red-arrow/doc/text/examples.md

```diff
@@ -0,0 +1,106 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Examples
+
+## Create table
+### From file
+```ruby
+table = Arrow::Table.load('data.arrow')
+table = Arrow::Table.load('data.csv', format: :csv)
+table = Arrow::Table.load('data.parquet', format: :parquet) # requires 'red-parquet' gem
+```
+### From Ruby hash
+Types will be detected automatically
+```ruby
+table = Arrow::Table.new('name' => ['Tom', 'Max'], 'age' => [22, 23])
+```
+### From String
+Suppose you have your data available via Http. Let's connect to demo Clickhouse db. See https://play.clickhouse.com/ for details
+```ruby
+params = {
+  query: "SELECT WatchID as watch FROM hits_v1 LIMIT 10 FORMAT Arrow",
+  user: "playground",
+  password: "clickhouse",
+  database: "datasets"
+}
+uri = URI('https://play-api.clickhouse.com:8443')
+uri.query = URI.encode_www_form(params)
+resp = Net::HTTP.get(uri)
+table = Arrow::Table.load(Arrow::Buffer.try_convert(resp))
+```
+
+## Filtering
+Uses concept of slicers in Arrow
+```ruby
+table = Arrow::Table.new(
+  'name' => ['Tom', 'Max', 'Kate'],
+  'age' => [22, 23, 19]
+)
+slicer = Arrow::Slicer.new(table)
+table.slice(slicer['age'] > 19)
+=> #<Arrow::Table:0x7fa38838c448 ptr=0x7fa3ad269f40>
+  name	age
+0	Tom 	 22
+1	Max 	 23
+
+table.slice(slicer['age'].in?(19..22))
+=> #<Arrow::Table:0x7fa3881cf998 ptr=0x7fa3a4bb5f30>
+  name	age
+0	Tom 	 22
+1	Kate	 19
+```
+Multiple slice conditions can be joined using and(`&`) / or (`|`) / xor(`^`) logical operations
+```ruby
+table.slice((slicer['age'] > 19) & (slicer['age'] < 23))
+=> #<Arrow::Table:0x7fa3882cc300 ptr=0x7fa3ad260b00>
+  name	age
+0	Tom 	 22
+```
+
+## Operations
+Arrow compute functions can be accessed through `Arrow::Function`
+```ruby
+add = Arrow::Function.find('add')
+add.execute([table['age'].data, table['age'].data]).value
```

Comment:
Ah, we should support `add.execute([table['age'], table['age']]).value` for this. Could you open a Jira issue for this?

Reply:
https://issues.apache.org/jira/browse/ARROW-14551